hw_2/src/main/java/com/epam/spring/homework2/beans/BeanE.java

```diff
@@ -0,0 +1,44 @@
+package com.epam.spring.homework2.beans;
+
+import com.epam.spring.homework2.validator.MyValidator;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+
+public class BeanE implements MyValidator {
+
+    private String name;
+    private double value;
+
+    public BeanE(String name, double value) {
+        this.name = name;
+        this.value = value;
+    }
+
+    @Override
+    public String toString() {
+        return "BeanE{" +
+                "name='" + name + '\'' +
+                ", value=" + value +
+                '}';
+    }
+
+    @PostConstruct
+    public void postConstructBeanE() {
+        System.out.println("postConstruct BeanE");
+    }
+
+    @PreDestroy
+    public void preDestroyBeanE() {
+        System.out.println("preDestroy BeanE");
+    }
+
+    @Override
+    public void validate() {
```

Comment:
validation method repeats for all beans

Could you please try find solution in order to move this method somewhere, so all the beans can reuse it
(one of possible solution - create parent class with this method and make all your beans extend that class)

Reply:
fixed